common/source/kernel/Thread.cpp

```diff
@@ -108,6 +108,13 @@ FileSystemInfo* Thread::getWorkingDirInfo(void)
   return working_dir_;
 }
 
+FileSystemInfo* getcwd()
+{
+  if (FileSystemInfo* info = currentThread->getWorkingDirInfo())
```

Comment:
what if currentThread == 0? where do you catch that?

Reply:
isn't that impossible? `currentThread` is only ever set from `Scheduler`'s `threads_` queue, which should never contain `nullptr`.

The old code had the `nullptr` check because `FakeThread* currentThread;` was set to it.